#! /bin/sh /usr/share/dpatch/dpatch-run

## 16_recordings.c-fix.dpatch by Joachim Wilke <joachim.wilke@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes segfault which occured when reaching the end of a replay

@DPATCH@
diff -urNad vdr-1.3.35/recording.c /tmp/dpep.eegJOQ/vdr-1.3.35/recording.c
--- vdr-1.3.35/recording.c	2005-10-31 13:27:58.000000000 +0100
+++ /tmp/dpep.eegJOQ/vdr-1.3.35/recording.c	2005-11-02 17:31:31.299956520 +0100
@@ -1342,7 +1342,7 @@
         // found a non existing file suffix
         }
      if (Open() >= 0) {
-        if (!record && Offset >= 0 && file->Seek(Offset, SEEK_SET) != Offset) {
+        if (!record && Offset >= 0 && file && file->Seek(Offset, SEEK_SET) != Offset) {
            LOG_ERROR_STR(fileName);
            return NULL;
            }
